x-pack/plugins/security_solution/server/endpoint/services/actions/action_list.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/core/server';
+import type { ActionDetails, ActionListApiResponse } from '../../../../common/endpoint/types';
+import type { SecuritySolutionRequestHandlerContext } from '../../../types';
+
+import {
+  getActionsForAgents,
+  getActionResponses,
+  getTimeSortedActionDetails,
+} from '../../utils/action_list_helpers';
+
+import {
+  categorizeActionResults,
+  categorizeResponseResults,
+  getActionCompletionInfo,
+  mapToNormalizedActionRequest,
+} from './utils';
+
+interface GetActionListParam {
+  actionTypes?: string[];
+  context: SecuritySolutionRequestHandlerContext;
+  elasticAgentIds?: string[];
+  endDate?: string;
+  logger: Logger;
+  page?: number;
+  pageSize?: number;
+  startDate?: string;
+  userIds?: string[];
+}
+
+export const getActionList = async ({
+  actionTypes,
+  context,
+  elasticAgentIds,
+  endDate,
+  logger,
+  page: _page,
+  pageSize,
+  startDate,
+  userIds,
+}: GetActionListParam): Promise<ActionListApiResponse> => {
+  const size = Math.floor(pageSize ?? 50 / 2);
+  const page = _page ?? 1;
+  const from = page <= 1 ? 0 : page * size - size + 1;
+
+  const data = await getActionDetailsList({
+    actionTypes,
+    context,
+    elasticAgentIds,
+    endDate,
+    from,
+    logger,
+    size,
+    startDate,
+    userIds,
+  });
+
+  return {
+    page,
+    pageSize,
+    startDate,
+    endDate,
+    userIds,
+    commands: actionTypes,
+    data,
+  };
+};
+
+export interface GetActionDetailsListParam extends GetActionListParam {
+  from?: number;
+  size?: number;
+}
+const getActionDetailsList = async ({
+  actionTypes,
+  context,
+  elasticAgentIds,
+  endDate,
+  from,
+  logger,
+  size,
+  startDate,
+  userIds,
+}: GetActionDetailsListParam): Promise<ActionDetails[]> => {
+  let actions;
+  let actionReqIds;
+
+  try {
+    // fetch actions with matching agent_id
+    const { actionIds, actionRequests } = await getActionsForAgents({
```

Comment:
I'm confused. where is `elasticAgentIds` coming from? We had agreed that all filtering query params were optional, so this could be undefined, but the utility you are calling here (`getActionsForAgents`) implies one that requires agent IDs to be defined.

Reply:
I would go with `getActionList()` or `getActions()`. `getActionsWithFilterOptions()` seems like it needs to have filter options defined to me.